src/WebJobs.Extensions.DurableTask/Storage/NameValidator.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Storage
+{
+    // This class is copied from the previous Azure Storage client SDKs
+    // The following logic may require updating over time as the Azure Storage team discourages client-side validation
+    // that may grow stale as the server evolves. See here: https://github.com/Azure/azure-sdk-for-js/issues/13519#issuecomment-822420305
+    internal static class NameValidator
```

Comment:
This doesn't exist anymore, so I honestly copied the decompiled validation logic 😅

Reply:
Given that the DTFx APIs actually create the storage account and related resources, it may make sense to leave it so that the task hub is not partially created.

I'm probably inclined to keep it, unless the name requirements start to shift. And if the validation here becomes a subset of what is actually allowed, it can always be removed later.